### DISCUSSION TITLE
Type mismatch does NOT create error during "compilation" time

### DISCUSSION THREAD

**Discussion by Magofoco** [MichalLytek/type-graphql#1208] [Q&A]
I would like to understand way there are no errors in this resolver even where there is a type mismatch.

I am saying that the Query should return an Int but it actually returns a String. If I execute it, it does complain. But is there a way to have the error at "compilation" time? Without me having to execute the query in the playground in order to see if there is a type error?

```
import { Int, Query, Resolver } from "type-graphql";

@Resolver()
export class MyResolver {
  @Query(() => Int)
  hello() {
    return "hello"
  }
}

```

**Comment by MichalLytek** [comment]
You can try this one:
https://github.com/borremosch/eslint-plugin-type-graphql

The best solution is to write an explicit return type, like:
```ts
  @Query(() => Int)
  hello(): string {
    return "hello"
  }
```

So you can see that `Int` is not `string`.
